Libraries/Network/FormData.js

```diff
@@ -65,21 +65,23 @@ class FormData {
     return this._parts.map(([name, value]) => {
       var contentDisposition = 'form-data; name="' + name + '"';
       var headers: Headers = {'content-disposition': contentDisposition};
-      if (typeof value === 'string') {
-        return {string: value, headers, fieldName: name};
-      }
 
       // The body part is a "blob", which in React Native just means
       // an object with a `uri` attribute. Optionally, it can also
       // have a `name` and `type` attribute to specify filename and
       // content type (cf. web Blob interface.)
-      if (typeof value.name === 'string') {
-        headers['content-disposition'] += '; filename="' + value.name + '"';
-      }
-      if (typeof value.type === 'string') {
-        headers['content-type'] = value.type;
+      if (typeof value === 'object') {
+        if (typeof value.name === 'string') {
+          headers['content-disposition'] += '; filename="' + value.name + '"';
+        }
+        if (typeof value.type === 'string') {
+          headers['content-type'] = value.type;
+        }
+        return {...value, headers, fieldName: name};
+      } else {
+        // Cast to string all other values
+        return {string: String(value), headers, fieldName: name};
```

Comment:
`else` after `return` is unnecessary. Any reason you moved the early bailout for the string case to down here?